docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,186 @@
+This napp has to keep consistency after reloading (re-read settings.py again, close and open rabbitmq sessions, with tests for these functionalities)
+This napp must support filtering specific REGEX messages via settings.py (a list of regex)
+
+Separating the message broker interaction into two distinct NApps makes the system architecture more modular and easier to manage. We can develop, deploy, and scale each NApp independently, providing flexibility and adaptability.
+Additionally, isolating the message broker functionality enhances fault tolerance. If one NApp experiences issues or failures, it does not directly impact the other NApp or the message broker. This isolation improves the overall robustness and reliability of the system.
+Creating two NApps to isolate the message broker simplifies the system architecture, enhances modularity, and improves fault tolerance, making developing, deploying, and maintaining messaging functionality easier.
+
+The topology_events NApp is responsible for two key functions:
+
+Publishing Serialized KytosEvents to a Message Broker:
+
+The primary responsibility of this NApp is to publish serialized KytosEvents to a message broker. Serialized KytosEvents contain structured data representing events within the Kytos topology.
+This involves converting KytosEvents into a serialized format compatible with the message broker's communication protocol.
+The NApp ensures the reliable delivery of these serialized events to the message broker, handling any necessary error handling or retry mechanisms.
+
+Filtering KytosEvents for Publication:
+
+Additionally, this NApp is responsible for filtering KytosEvents before publishing them to the message broker.
+Filtering may involve selecting specific types of events, applying rules or criteria to determine which events are relevant for publication, or transforming events into a different format as downstream consumers require.
+The NApp optimizes communication by only transmitting relevant data to the message broker by filtering events, reducing unnecessary network traffic and processing overhead.
+
+Overall, this NApp acts as a bridge between Kytos' internal message bus and the message broker, facilitating the exchange of events between the internal network infrastructure and external systems. It guarantees the serialization, filtering, and reliable transmission of relevant events to the message broker for downstream applications or service consumption.
+
+The Broadcasting Events Napp (topology_events) must be able to utilize any replaceable message transportation system. This requirement underscores the necessity for flexibility and adaptability in the system's architecture. Consequently, the Broadcasting Events Napp and the Events Consumer Napp should be isolated into microservices napps, distinct from the Message Queue Producer and Consumer Napps. This architectural decision offers several benefits:
```

Comment:
> and the Events Consumer Napp should be isolated into microservices napps

There's no such thing as a microservice NApp. A NApp can be a "microapplication" with a defined responsiblity and you can certainly also run a kytos instance with just a single NApp acting as a sort of microservice, and for certain cases this might work when you're basically just using it to reuse `Starlette` and a NApp tiny application, but Kytos-ng is a plugin monolithic architecture. 

What you're saying about microservice generally is true and makes sense, but we NApps aren't microservices and won't be in the short or medium term, the platform itself shares many memory data structures, app instance, db instance and so on. I'm just reiterating this here just so we don't get confused or future readers don't get confused calling a NApp a microservice (by the usual software industry definition)